src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,47 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: String) -> Result<(), String> {
+    let mut devfile = OpenOptions::new()
+        .read(true)
+        .open(&devpath)
+        .map_err(|_the_io_error| format!("Error opening device: {}", _the_io_error))?;
+
+    let bda = BDA::load(&mut devfile).unwrap().unwrap();
```

Comment:
Ok! Now the outermost unwrap() here is our next target. It means that there is definitely no Stratis header on the device, hence no BDA. We're going to treat no BDA as an error for purposes of this script.

Reply:
Resolved.